### PR TITLE
feat: Add CollectionCuration & ItemCuration

### PR DESCRIPTION
**Pull request by juanmahidalgo**
## Description
For #1725, we'll need to have `ItemCuration` alongside `CollectionCuration`. This PR renames the current `Curation` to `CollectionCuration` and add the new endpoint to the `builderAPI`

### CODE REVIEW

File: src/components/ItemEditorPage/TopPanel/TopPanel.tsx [hunk 1]
```diff
@@ -3,13 +3,14 @@ import { Button, Loader } from 'decentraland-ui'
...
 import { Collection, CollectionType } from 'modules/collection/types'
-import { Curation, CurationStatus } from 'modules/curation/types'
+import { CollectionCuration, CurationStatus } from 'modules/collectionCuration/types'
 import { getCollectionType, hasReviews } from 'modules/collection/utils'
 import CollectionProvider from 'components/CollectionProvider'
 import JumpIn from 'components/JumpIn'
 import RejectionModal from './RejectionModal'
 import { RejectionType } from './RejectionModal/RejectionModal.types'
 import { ButtonType, Props, State } from './TopPanel.types'
+import { ItemCuration } from 'modules/itemCuration/types'
```
Comment (nicosantangelo) [inline src/components/ItemEditorPage/TopPanel/TopPanel.tsx:13]: can we move this import below `modules/collectionCuration/types`?

File: src/modules/collectionCuration/actions.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+export const approveCollectionCurationRequest = (collectionId: string) => action(APPROVE_COLLECTION_CURATION_REQUEST, { collectionId })
+export const approveCollectionCurationSuccess = (collectionId: string) => action(APPROVE_COLLECTION_CURATION_SUCCESS, { collectionId })
+export const approveCollectionCurationFailure = (collectionId: string, error: string) => action(APPROVE_COLLECTION_CURATION_FAILURE, { collectionId, error })
+
+export type ApproveCollectionCurationRequestAction = ReturnType<typeof approveCollectionCurationRequest>
+export type ApproveCollectionCurationSuccessAction = ReturnType<typeof approveCollectionCurationSuccess>
+export type ApproveCollectionCurationFailureAction = ReturnType<typeof approveCollectionCurationFailure>
+
+// Reject Collection CollectionCuration
```
Comment (nicosantangelo) [inline src/modules/collectionCuration/actions.ts:61]: We might want to just say:
```ts
'[XXX] Approve CollectionCuration'
// or
'[XXX] Approve Collection Curation'
```
and
```ts
// Reject Collection Curation
```


here. Same for the other actions. Wdyt?

File: src/modules/collectionCuration/types.ts [hunk 3]
```diff
@@ -4,10 +4,13 @@ export enum CurationStatus {
   REJECTED = 'rejected'
 }
```
Comment (nicosantangelo) [inline src/modules/collectionCuration/types.ts:6]: I don't want you to have to move everything again, but what do you think of this:
- Encompassing `collectionCuration` inside `/modules/curation` so it looks like:
```
/modules/curation
  collectionCuration/
  itemCuration/
```

- add top level logic in curation:
```
/modules/curation
  types.ts # contains CurationStatus and BaseCuration for example, useful for both
  collectionCuration/
  itemCuration/
```

File: src/modules/collectionCuration/sagas.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  pushCollectionCurationSuccess,
+  PUSH_COLLECTION_CURATION_REQUEST,
+  rejectCollectionCurationFailure,
+  RejectCollectionCurationRequestAction,
+  rejectCollectionCurationSuccess,
+  REJECT_COLLECTION_CURATION_REQUEST
+} from './actions'
+import { CollectionCuration, CurationStatus } from './types'
+
+export function* curationSaga(builder: BuilderAPI) {
```
Comment (nicosantangelo) [inline src/modules/collectionCuration/sagas.ts:30]: this saga should maybe be called `collectionCurationSaga` right?

File: src/modules/collectionCuration/reducer.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+  | PushCollectionCurationSuccessAction
+  | PushCollectionCurationFailureAction
+  | ApproveCollectionCurationRequestAction
+  | ApproveCollectionCurationSuccessAction
+  | ApproveCollectionCurationFailureAction
+  | RejectCollectionCurationRequestAction
+  | RejectCollectionCurationSuccessAction
+  | RejectCollectionFailureSuccessAction
+
+export function curationReducer(state: CurationState = INITIAL_STATE, action: CurationReducerAction): CurationState {
```
Comment (nicosantangelo) [inline src/modules/collectionCuration/reducer.ts:65]: this reducer should maybe be called `collectionCurationReducer` right?

**Review by nicosantangelo — COMMENTED** [review]
Gr8 W0rk

File: src/components/ItemEditorPage/TopPanel/TopPanel.tsx (hunk 1 above)
Comment (juanmahidalgo) [reply src/components/ItemEditorPage/TopPanel/TopPanel.tsx]: For sure! moved

File: src/modules/collectionCuration/actions.ts (hunk 2 above)
Comment (juanmahidalgo) [reply src/modules/collectionCuration/actions.ts]: Absolutely, updated!

File: src/modules/collectionCuration/sagas.ts (hunk 4 above)
Comment (juanmahidalgo) [reply src/modules/collectionCuration/sagas.ts]: Yeah! updated to `collectionCurationSaga` :)

File: src/modules/collectionCuration/reducer.ts (hunk 5 above)
Comment (juanmahidalgo) [reply src/modules/collectionCuration/reducer.ts]: Yup, updated to `collectionCurationReducer`!

**Review by nicosantangelo — APPROVED** [review]
LGTM

**Pull request merged**